lib/jormungandr/test/unit/Cardano/Wallet/Jormungandr/BinarySpec.hs

```diff
@@ -200,6 +236,114 @@ spec = do
                 `shouldThrow` userException msg
 
         it "throws when encoding an address of invalid length" $ do
-            let msg = "Address has unexpected length 1: Address {unAddress = \"0\"}"
+            let msg = "Address has unexpected length 1: \
+                \Address {unAddress = \"0\"}"
             evaluate (runPut $ putAddress $ Address "0")
                 `shouldThrow` userException msg
+
+        it "decode (encode address) === address" $ property $ \addr -> do
+                let bytes = (runPut $ putAddress addr)
+                let addr' = try' $ runGet getAddress bytes
+                addr' === (Right addr)
+
+        it "decode (encode tx) === (validate tx)" $
+            property $ \(SignedTx signedTx) -> do
+                let bytes = try' (runPut $ putSignedTx signedTx)
+                let tx' = try' . unMessage . runGet getMessage =<< bytes
+                tx' === (Right signedTx)
```

Comment:
Why not simply `runPut` and `runGet`? Since you're testing happy path.. let it throw in case of unexpected error. `unsafePerformIO` is not really something we want to use when we can avoid it and here clearly, there's no need. 

Note that (since you mentioned it somewhere) using `catch` and a `monadicIO` property isn't much more work:

```hs
property $ \(SignedTx signedTx) -> monadicIO $ liftIO $ do
    return (runPut ...) `catch` (\MyException -> ....)
```

Reply:
🤔 Don't we lose failing-test-case-output without `===` ?

I guess this is still a roundtrip so we would be using like `assert (tx' == tx)` still?

Reply:
Hmmm.. why would we loose them? In case of failure, we can call something like `expectationFailure` with a clear error message.

Reply:
``shouldBe`` seemed to work well, except for hanging for ~40 s  when printing large outputs. I suspect it's trying to find the minimal-edit-distance.

So went with:
```haskell
                if tx' == Right signedTx
                then return ()
                else expectationFailure $
                    "tx /= decode (encode tx) == " ++ show tx'
```

resulting in the following output:

<img width="1440" alt="Skärmavbild 2019-07-02 kl  19 18 36" src="https://user-images.githubusercontent.com/304423/60532710-32d5a800-9cfe-11e9-96fb-8c6077587723.png">